sisyphus/simple_linux_utility_for_resource_management_engine.py

```diff
@@ -86,16 +86,21 @@ def system_call(self, command, send_to_stdin=None):
         """
         if self.gateway:
             escaped_command = [shlex.quote(s) for s in command]  # parameters need to be shell safe when sending via ssh
-            system_command = ["ssh", "-x", self.gateway] + [" ".join(["cd", os.getcwd(), "&&"] + escaped_command)]
+            system_command = ["ssh", "-x", self.gateway, "-o", "BatchMode=yes"] + [
+                " ".join(["cd", os.getcwd(), "&&"] + escaped_command)
+            ]
         else:
             # no gateway given, skip ssh local
             system_command = command
 
         logging.debug("shell_cmd: %s" % " ".join(system_command))
-        p = subprocess.Popen(system_command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
         if send_to_stdin:
             send_to_stdin = send_to_stdin.encode()
-        out, err = p.communicate(input=send_to_stdin, timeout=30)
+        try:
+            p = subprocess.run(system_command, input=send_to_stdin, capture_output=True, timeout=30)
+        except subprocess.TimeoutExpired:
```

Comment:
If we catch the `subprocess.TimeoutExpired` exception here and return, then the whole logic about retrying in `submit_helper` https://github.com/rwth-i6/sisyphus/blob/51db853857ab53b43ebb2f91b6e1347b81193bd7/sisyphus/simple_linux_utility_for_resource_management_engine.py#L239-L246 is obsolet. 
If this was intentional, then maybe also remove this logic? and `gs.WAIT_PERIOD_SSH_TIMEOUT` would be ignored then

Reply:
Ah so it is actually handled. That was my earlier question about this. So then I would not catch it. Unless there is good reason to change this old behavior about `TimeoutExpired`.

Reply:
Yes... I hadn't noticed that it would be catched later in the call stack. See https://github.com/rwth-i6/sisyphus/pull/196.